pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -292,35 +298,24 @@ func (c *Cluster) startProvisioningOverPVCs(config *provisionConfig) {
 			}
 		}
 
-		// Update the orchestration status of this pvc to the starting state
-		status := OrchestrationStatus{Status: OrchestrationStatusStarting, PvcBackedOSD: true}
-		c.updateOSDStatus(osdProps.crushHostname, status)
-
 		// Skip OSD prepare if deployment already exists for the PVC
-		listOpts := metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%s,%s=%s",
-			k8sutil.AppAttr, AppName,
-			OSDOverPVCLabelKey, dataSource.ClaimName,
-		)}
-
-		osdDeployments, err := c.context.Clientset.AppsV1().Deployments(c.clusterInfo.Namespace).List(listOpts)
-		if err != nil {
-			config.addError("failed to check if OSD daemon exists for pvc %q. %v", osdProps.crushHostname, err)
-			continue
-		}
-
-		if len(osdDeployments.Items) != 0 {
+		if osdDeployment, ok := existingDeployments[dataSource.ClaimName]; ok {
 			logger.Infof("skip OSD prepare pod creation as OSD daemon already exists for %q", osdProps.crushHostname)
-			osds, err := c.getOSDInfo(&osdDeployments.Items[0])
+			osds, err := c.getOSDInfo(osdDeployment)
 			if err != nil {
 				config.addError("failed to get osdInfo for pvc %q. %v", osdProps.crushHostname, err)
 				continue
 			}
 			// Update the orchestration status of this pvc to the completed state
-			status = OrchestrationStatus{OSDs: osds, Status: OrchestrationStatusCompleted, PvcBackedOSD: true}
+			status := OrchestrationStatus{OSDs: osds, Status: OrchestrationStatusCompleted, PvcBackedOSD: true}
```

Comment:
I guess you moved L295-L298 to L315-L318 because OSDStatus isn't needed to update at all if osdDeployment already exists. If so, we can remove the above status update.

Reply:
Yes, that's why I moved it, no need for the status update if the OSD is skipping the prepare job. Are you just confirming the desired behavior, or are you suggesting another place to remove the status?

Reply:
I meaned suggestion. I should change request than comment, sorry. My suggestion is removing status change in L309-L311.

Reply:
The status change on 310-311 is actually required. It sets the status to completed so that the operator will be able to process the OSD deployments in the next phase of reconcile after the prepare jobs are all started.